### DISCUSSION TITLE
[2.9.1] - 2023-03-02

### DISCUSSION THREAD

**Discussion by srozum** [srozum/film_camera_tester#13] [Announcements]
Version for photosensors PIC0103SL with inverted output signal, instead of normal PIC0903SL (SIC553-04 old part name)

<hr /><em>This discussion was created from the release <a href='https://github.com/srozum/film_camera_tester/releases/tag/2.9.1'>[2.9.1] - 2023-03-02</a>.</em>

**Comment by srozum** [comment]
It's about Photodiodes, or Photo IC (photodiode with incorporated amplifier and schmitt trigger) used in Sensors #1 and #2.

Originally, those "Laser Sensor" modules available everywhere were equipped with Photo IC called SIC553-04 made by Kodenshi Corp., but those were discontinued parts.

Recently all those modules on AliExpress and Amazon were updated with little note:
- output high level when receive laser signal;
- output low level when not receive laser signal.

Product photos and module schematics all the same, but output signal is inverted.

I did some research an found that Kodenshi updated their line of photosensors with parts PIC0103SL and PIC0903SL, which are according to datasheets are exactly the same as SIC553-04 but one has normal open-collector output and another has inverted output.

I believe that PIC0103SL is what currently used in those modules. While PIC0903SL is a new version of SIC553-04.

Either should work in Tester using current PCB designs, however if you got photo IC with inverted output the readings will be reversed... means that Tester will measure dark time.

Firmware v2.9.1 should be used if you bought modules equipped with inverted sensors.

**Comment by Pitichinaccio** [comment]
Hello Serhiy,

I also had the problem with the inverting types of photodiodes. I flashed to 2.9.1 now and I can confirm that it is working fine.
Many thanks for providing this alternative code. 

Bernd

**Comment by ic-racer** [comment]
I ordered my sensors from AliExpress in April of '23 and got the older versions so I'm running firmware 2.9. So, the older sensors are still out there from some suppliers. Using a six-prong plug on my software flasher, I found changing the firmware between the two very easy.

**Comment by Surreal62** [comment]
Just getting materials in for building the tester and ordered a few from Aliexpress but no idea what photocells they are.
Firmware is as ic-racer mentioned is very easy to flash with a USBasp and AVRDUDESS.

**Comment by Surreal62** [comment]
Tester functioning on frimware 2.9 and issues previously encountered rectified. 
However, I have the inverted output photo sensors so need to use rev.2.9.1 firmware and I have flashed it to three different Nanos and the tester will not run. Powers up to the display showing the two lines of blocks.
Reverting to rev. 2.9 all three work fine, bar measuring dark time - which it does very accurately! 👍👍
No flashing errors show when uploading rev.2.9.1
Any help greatly appreciated
